client-side/test/stories/Table/Table.test.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import TableComponent from '../../../src/stories/table/TableComponent';
+
+describe('TableComponent', () => {
+  const mockData1 = {
+    dataObject: {
+      headers: ['id', 'name', 'age', 'city'],
+      rows: [
+        { id: 1, name: 'יונתן', age: 25, city: 'תל אביב' },
+        { id: 2, name: 'מיכל', age: 30, city: 'ירושלים' },
+      ],
+    },
+    widthOfTable: "60%",
+    widthOfColums: [150, 150, 150, 150],
+  };
+
+  const mockData2 = {
+    dataObject: {
+      headers: ['id', 'name', 'profession', 'age', 'city'],
+      rows: [
+        { id: 1, name: 'יונתן', profession: 'אנגלית', age: 25, city: 'תל אביב' },
+        { id: 2, name: 'מיכל', profession: 'חשבון', age: 30, city: 'ירושלים' },
+      ],
+    },
+    widthOfTable: "80%",
+    widthOfColums: [150, 150, 150, 150, 150],
+  };
+
+  test('renders the table with data1 correctly', () => {
+    render(<TableComponent {...mockData1} />);
```

Comment:
add also test for case that should failed - maybe wrong data or something like that. to see that the application is not crashed